components/base-node.js

```diff
@@ -65,7 +79,32 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
```

Comment:
I'm curious why the 'push' function is being defined above on line 62/76.  To automatically create an array if there isn't one already?